File: Bit_manipulations/InterestingArray.java
public class Solution {
    public String solve(ArrayList<Integer> A) {
        int xor=0;
        int i,n=A.size();
        int numodds=0;
        for(i=0;i<n;i++){
           
            if(A.get(i)%2==1){
                numodds+=1;
                 xor=xor^A.get(i);
            }
        }
        if(xor==0 || xor%2==0 || numodds%2==0){
            return "Yes";
        }
        else{
            return "No";
        }
        //return "No";
        
}
}

File: Bit_manipulations/InterestingArray.py
class Solution:
    # @param A : list of integers
    # @return a strings
    def solve(self, A):
        n=len(A)
        even=0
        odd=0
        for i in A:
            if(i%2==0):
                even+=1
            else:
                odd+=1
        if(odd%2==0):
            return ("Yes")
        else:
            return ("No")
